ipaclient/install/client.py

```diff
@@ -2470,18 +2470,21 @@ def sync_time(options, fstore, statestore):
     # disable other time&date services first
     timeconf.force_chrony(statestore)
 
-    logger.info('Synchronizing time')
-
-    if not options.ntp_servers:
+    if not options.ntp_servers and not options.ntp_pool:
         ds = discovery.IPADiscovery()
         ntp_servers = ds.ipadns_search_srv(cli_domain, '_ntp._udp',
                                            None, break_on_first=False)
-    else:
-        ntp_servers = options.ntp_servers
+        if not ntp_servers and not options.unattended:
+            timeconf.get_time_source(options)
```

Comment:
I don't like functions that modify an argument as a side effect. It's too magic and hard to understand. How do you like:

```
options.ntp_servers, options.ntp_pool = timeconf.get_time_source()
```

It's explicit and the reader immediately understands what the function is doing.

Reply:
Understand.